raft/raft.go

```diff
@@ -242,6 +246,7 @@ func (r *raft) bcastHeartbeat() {
 		}
 		r.sendHeartbeat(i)
 		r.prs[i].resume()
+		r.prs[i].ins.reset()
```

Comment:
I am not quite comfortable with this one too. 

But here is the problem:
If there is no resetting mechanism, the progress might get stuck. If the leader sends out msgApps to a follower successfully and the window is full, but all the msgAppResps are lost. There is no error reporting to the leader since the sending is successful. And the following msgHeartbeats are all good. Then the progress of that follower gets stuck. We can assume that the failure of msgAppResp can be reported at the follower side, but we actually have no mechanism to retry msgAppResp.